packages/core/contracts/Escrow.sol

```diff
@@ -228,63 +223,46 @@ contract Escrow is IEscrow, ReentrancyGuard {
         require(aggregatedBulkAmount < BULK_MAX_VALUE, 'Bulk value too high');
 
         if (balance < aggregatedBulkAmount) {
-            return bulkPaid;
+            return false;
         }
 
         _storeResult(_url, _hash);
 
         (
+            uint256[] memory finalAmounts,
             uint256 reputationOracleFee,
             uint256 recordingOracleFee
         ) = finalizePayouts(_amounts);
 
-        uint256[] memory _amountsPaid = new uint256[](_recipients.length);
         for (uint256 i = 0; i < _recipients.length; ++i) {
-            uint256 amount = finalAmounts[i];
-            if (amount == 0) {
-                continue;
-            }
-            finalAmounts[i] = 0;
-            _safeTransfer(_recipients[i], amount);
-            _amountsPaid[i] = amount;
+            _safeTransfer(_recipients[i], finalAmounts[i]);
         }
 
-        delete finalAmounts;
-
         _safeTransfer(reputationOracle, reputationOracleFee);
         _safeTransfer(recordingOracle, recordingOracleFee);
 
         bulkPaid = true;
+
         balance = getBalance();
-        if (bulkPaid) {
-            if (status == EscrowStatuses.Pending) {
-                status = EscrowStatuses.Partial;
-                remainingSolutions = remainingSolutions.sub(_recipients.length);
-            }
-            if (
-                balance > 0 &&
-                status == EscrowStatuses.Partial &&
-                remainingSolutions == 0
-            ) {
-                _safeTransfer(canceler, balance);
-                status = EscrowStatuses.Paid;
-            }
-            if (balance == 0 && status == EscrowStatuses.Partial) {
-                status = EscrowStatuses.Paid;
-            }
+        if (balance == 0) {
+            status = EscrowStatuses.Paid;
+        } else {
+            status = EscrowStatuses.Partial;
         }
+
         emit BulkTransfer(
             _txId,
             _recipients,
-            _amountsPaid,
+            finalAmounts,
             status == EscrowStatuses.Partial
```

Comment:
Why dont we just emit the status here instead of a boolean?

Reply:
boolean field is more descriptive as an argument of this event emission.